### PR TITLE
typescript: extract webdriver creation to separate factory

### PR DESCRIPTION
**Pull request by dbjorge**
#### Description of changes

Fixes the issue with upgrading the chromedriver package by using a more reliable means of delegating to the Azure Pipelines ChromeWebDriver variable.

As part of this, also:
* Extracts webdriver creation to a separate factory file, to keep the main test file simpler.
* Adds support for using Firefox too, based on user study feedback and for consistency with C# sample

Starting as a draft to see if a suspicious looking local test failure repros on the build machine.

#### Pull request checklist

- [x] If this PR addresses an existing issue, it is linked: Further builds on #48
- [ ] `yarn test` passes in all affected samples
- [x] Added any applicable tests

### CODE REVIEW

**Comment by dbjorge** [comment]
Blocked on https://github.com/DefinitelyTyped/DefinitelyTyped/pull/37854

**Comment by dbjorge** [comment]
Unblocked by @types/selenium-webdriver v4.0.3

File: typescript-selenium-webdriver-sample/tests/webdriver-factory.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import * as webdriver from 'selenium-webdriver';
+import * as chrome from 'selenium-webdriver/chrome';
+import * as firefox from 'selenium-webdriver/firefox';
+import * as path from 'path';
+
+export function createWebdriverFromEnvironmentVariableSettings(): webdriver.ThenableWebDriver {
+    // Selenium WebDriver implementations generally require that you use a version of the webdriver
+    // that exactly matches the version of the browser it is driving.
+    //
+    // In our Azure Pipelines build agents, using the pre-installed versions of the webdrivers that
```
Comment (smoralesd) [inline typescript-selenium-webdriver-sample/tests/webdriver-factory.ts]:
```
    // In our Azure Pipelines build agents, using the pre-installed versions of the webdrivers that
    // match the pre-installed versions of the browsers.
```
not sure if I'm missing something here but the phrasing seems a little off to me

Comment (dbjorge) [reply typescript-selenium-webdriver-sample/tests/webdriver-factory.ts]: agreed, fixed

File: typescript-selenium-webdriver-sample/azure-pipelines.yml [hunk 2]
```diff
@@ -27,44 +27,53 @@
...
   #
   # Our sample uses Yarn for this, but if your project uses NPM instead, that's also fine.
-  #
-  # Note that we add a specific command line option here to tell the "chromedriver" package to
-  # use the preinstalled version of the Chrome webdriver that matches the preinstalled version of
-  # Chrome on the hosted Microsoft build agent. The build agent comes with a "ChromeWebDriver"
-  # environment variable pre-set to the location of the preinstalled chromedriver.exe. You would
-  # typically omit this argument when running "yarn install" locally, since most users won't have
-  # chromedriver preinstalled like the build agent will.
-  - script: yarn install --frozen-lockfile --chromedriver_filepath=$(ChromeWebDriver) # or "npm install --ci"
```
[the remark this answers is not in the document]
Comment (dbjorge) [reply typescript-selenium-webdriver-sample/azure-pipelines.yml]: 2 main reasons:
* It wasn't very reliable: it sometimes still pulled an older version than it should have, for reasons we never root caused
* geckodriver doesn't support a similar parameter, and we wanted their handling to look consistent

The logic in `webdriver-factory.ts` replaces this; it's now the place where we use the environment variables (for both webdrivers)

**Pull request merged**